run-clang-format.py

```diff
@@ -25,20 +25,35 @@
 style = sys.argv[4]
 print("Style: " + style)
 
-def runclangformat(filepath):
-    print("Checking: " + filepath)
+class FileInfo:
+    def __init__(self, file_path, parent_folder):
+        self.file_path = file_path
+        self.parent_folder = parent_folder
+
+
+def runclangformat(fileInfo):
+    filepath = fileInfo.file_path
+    if fileInfo.parent_folder:
+        print("Checking: " + fileInfo.parent_folder)
+
     # See: https://stackoverflow.com/questions/22866609/can-clang-format-tell-me-if-formatting-changes-are-necessary
     proc = subprocess.Popen("clang-format -style=" + style + " " + filepath + " | diff -u --color=always " + filepath + " -", shell=True)
     if proc.wait() != 0:
         failedfiles.append(filepath)
 
 def collectfiles(dir, exclude, exts):
     collectedfiles = []
+    unique_folders = set()
```

Comment:
`folders` is very Windows terminology. May be `unique_dirs`?

Reply:
Make sense